fastlane_core/lib/fastlane_core/cert_checker.rb

```diff
@@ -200,5 +200,18 @@ def self.sha1_fingerprint(path)
       UI.error(error)
       UI.user_error!("Error parsing certificate '#{path}'")
     end
+
+    def self.certificate_key_match?(cert_path, key_path)
+      cert_md5 = `openssl x509 -inform der -noout -modulus -in '#{cert_path}' | openssl md5`.strip
+      key_md5_rsa = `openssl rsa -noout -modulus -in '#{key_path}' | openssl md5`.strip
+      temp_key_path = File.join(File.dirname(key_path), "-temp-key")
+      `openssl pkcs12 -in #{key_path} -nocerts -passin pass: -nodes -out #{temp_key_path}`
+      key_md5_pkcs12 = `openssl rsa -noout -modulus -in #{temp_key_path} | openssl md5`.strip
+      `rm #{temp_key_path}`
```

Comment:
Ruby has a built-in `OpenSSL` class, would it be possible to use this instead of shelling out to the command line?